src/core/vdom/create-component.js

```diff
@@ -208,6 +208,10 @@ function destroy (vnode: MountedComponentVNode) {
   }
 }
 
+// return three possible result
+// 1. Ctor -> component constructor if resolved successfully
+// 2. undefined -> component is pending
+// 3. null -> component rejected
 function resolveAsyncComponent (
```

Comment:
hmm maybe have some sort of enum integer consts like `const PENDING = 1` and check if `res === PENDING`